template/androidApp/src/main/java/io/bloco/template/MainActivity.kt

```diff
@@ -18,8 +37,50 @@ class MainActivity : ComponentActivity() {
     @InstallIn(ActivityComponent::class)
     interface ViewModelFactoryProvider : BaseViewModelFactoryProvider
 
+    private val viewModel: MainActivityViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
-        setContent { TemplateTheme { TemplateNaveHost() } }
+
+        var uiState: MainActivityUiState by mutableStateOf(Loading)
+
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.uiState
+                    .onEach {
+                        uiState = it
+                    }.collect()
+            }
+        }
+
+        // Keep the splash screen on-screen until the UI state is loaded. This condition is
+        // evaluated each time the app needs to be redrawn so it should be fast to avoid blocking
+        // the UI.
+        splashScreen.setKeepOnScreenCondition {
+            when (uiState) {
+                Loading -> true
+                Success -> false
+            }
+        }
```

Comment:
I feel this code should be close to the `val splashScreen = installSplashScreen()` code.

Reply:
val splashScreen = installSplashScreen() needs to be before super.onCreate() so it can replace the theme.
this one always needs to be after... but I guess I can move it a few instructions above, would have to test it